src/ControlSystem/Metafunctions.hpp

```diff
@@ -44,5 +44,45 @@ using control_systems_with_measurement_t =
     typename control_systems_with_measurement<ControlSystems,
                                               Measurement>::type;
 /// @}
+
+/// Given a measurement, obtain a list of its submeasurements (i.e.,
+/// `Measurement::submeasurements`).
+/// @{
+template <typename Measurement>
+struct submeasurements {
+  using type = typename Measurement::submeasurements;
+};
+
+template <typename Measurement>
+using submeasurements_t = typename submeasurements<Measurement>::type;
+/// @}
+
+namespace detail {
+template <typename Submeasurement, typename ControlSystems>
+struct interpolation_target_tags_for_submeasurement {
+ private:
+  using declared_type =
+      typename Submeasurement::template interpolation_target_tag<
+          ControlSystems>;
+
+ public:
+  using type = tmpl::conditional_t<std::is_same_v<declared_type, void>,
+                                   tmpl::list<>, tmpl::list<declared_type>>;
```

Comment:
Just a question. Why does the conforming type have to specify void if it's just going to end up as an empty list anyways? Why can't it just originally be an empty list?

Reply:
The alias isn't a list, just a type.